spec/src/modules/search.js

```diff
@@ -298,14 +298,17 @@ describe(`ConstructorIO - Search${bundledDescriptionSuffix}`, () => {
       });
 
       search.getSearchResults(query, { section, hiddenFields }, {}).then((res) => {
+
         const requestedUrlParams = helpers.extractUrlParamsFromFetch(fetchSpy);
 
+        const resultWithTestField = res.response.results.find(result => result.data.testField);
+
         expect(res).to.have.property('request').to.be.an('object');
         expect(res).to.have.property('response').to.be.an('object');
         expect(res).to.have.property('result_id').to.be.an('string');
         expect(res.request.fmt_options.hidden_fields).to.eql(hiddenFields);
         expect(requestedUrlParams.fmt_options).to.have.property('hidden_fields').to.eql(hiddenFields);
-        expect(res.response.results[0].data).to.have.property('testField').to.eql('hiddenFieldValue');
+        expect(resultWithTestField).to.be.not.null.to.eql('hiddenFieldValue');
```

Comment:
I think we actually need to change this. res.response.results.find returns undefined when nothing is found. And undefined !== null.

We will probably have to use something else other than to.be.not.null because that will always be true at the moment.

Reply:
Good eye, added better tests for that case.